src/solvers/sweep_solve.jl

```diff
@@ -0,0 +1,40 @@
+
+default_region_callback(problem; kws...) = nothing
+
+default_sweep_callback(problem; kws...) = nothing
+
+function default_sweep_printer(problem; outputlevel, sweep, nsweeps, kws...)
+  if outputlevel >= 1
+    println("Done with sweep $sweep/$nsweeps")
+  end
+end
+
+function sweep_solve(
+  sweep_iterator;
+  outputlevel=0,
+  region_callback=default_region_callback,
+  sweep_callback=default_sweep_callback,
+  sweep_printer=default_sweep_printer,
+  kwargs...,
+)
+  for (sweep, region_iter) in enumerate(sweep_iterator)
+    for (region, region_kwargs) in region_tuples(region_iter)
+      region_callback(
+        problem(region_iter);
```

Comment:
I noticed this gets passed the problem while `sweep_callback` and `sweep_printer` get passed the `region_iter`, is there a logic behind that? I would lean towards all of them getting passed the `region_iter`, and there could be a layer that unwraps the problem as needed.